org-roam.el

```diff
@@ -134,6 +134,27 @@ space-delimited strings.
            (symbol)))
   :group 'org-roam)
 
+(defcustom org-roam-tag-sources '(prop)
+  "Sources to obtain tags from.
+
+It should be a list of symbols representing any of the following
+extraction methods:
+
+1. 'prop: This extracts tags from the =#+ROAM_TAGS=
+   property.  Tags are space delimited, and can be multi-word
+   using double quotes.
+
+2. 'all-directories: All sub-directories relative to
+   =org-roam-directory= are extracted as tags.  That is, if a file
+   is located at relative path =foo/bar/file.org=, the file will
+   have tags =foo= and =bar=.
+
+3. 'last-directory: Extracts the last directory relative to
+   =org-roam-directory= as the tag.  That is, if a file is located
+   at relative path =foo/bar/file.org=, the file will have tag
+   =bar=."
+  :type '(repeat symbol))
```

Comment:
```suggestion
   :group 'org-roam
  :type '(set (const :tag "#+ROAM_TAGS" prop)
              (const :tag "sub-directories" all-directories)
              (const :tag "parent directory" last-directory)))
```
This will give users the specific symbol options in the customization system.